### PR TITLE
Build script improvements

### PR DESCRIPTION
**Pull request by modmuss50**
- Gradle 8.1
- Move more to filament
- Task caching should be a lot better
- Configuration caching enabled

### CODE REVIEW

**Comment by thecatcore** [comment]
Bump filament version?

**Comment by RawDiamondMC** [comment]
Update to gradle 8.0.2?

**Review by thecatcore — APPROVED** [review]

File: filament/src/main/java/net/fabricmc/filament/task/enimga/ConvertMappingsTask.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
+package net.fabricmc.filament.task.enimga;
```
Comment (Juuxel) [inline filament/src/main/java/net/fabricmc/filament/task/enimga/ConvertMappingsTask.java]: "enimga"

File: gradle.properties [hunk 2]
```diff
@@ -1,16 +1,18 @@
 # Done to increase the memory available to gradle.
 org.gradle.jvmargs=-Xmx1G
+org.gradle.parallel=true
```
Comment (Juuxel) [inline gradle.properties:3]: Does this matter? To my knowledge, Yarn doesn't have subprojects. This flag only affects multiproject builds. (See [this discord discussion](https://discord.com/channels/507304429255393322/592749519171616796/1021440447144546384))

**Review by Juuxel — CHANGES_REQUESTED** [review]

File: gradle.properties (hunk 2 above)
Comment (modmuss50) [reply gradle.properties:3]: No, I double checked and it doesnt, no real harm in setting it tho.

**Pull request merged**